samples/excel/89-preview-apis/events-worksheet-changed.yaml

```diff
@@ -0,0 +1,177 @@
+order: 9
+id: excel-events-worksheet-changed
+name: Events - Worksheet changed
+description: Add event handlers on the data changed and worksheet added events
+host: EXCEL
+api_set:
+    ExcelApi: 1.7
+script:
+    content: |+
+        $("#setup").click(() => tryCatch(setup));
+        $("#egister-on-data-changed-handler").click(() => tryCatch(registerOnDataChangedHandler));
+        $("#data-changed").click(() => tryCatch(changeData));
+        $("#register-on-add-handler").click(() => tryCatch(registerOnAddHandler));
+        $("#add-worksheet").click(() => tryCatch(addWorksheet));
+
+        async function registerOnDataChangedHandler() {
+            await Excel.run(async (context) => {
+                let sheet = context.workbook.worksheets.getItem("Sample");
+                sheet.onDataChanged.add(onDataChange);
+
+                await context.sync();
+
+                OfficeHelpers.UI.notify("A handler has been registered for the onDataChanged event.");
+            });
+        }
+
+        async function changeData() {
```

Comment:
I don't think you should programmatically change the data or add the worksheet. That might give the impression that the events are only triggered programmatically. but 99% of the time these events will be triggered by manual action of the user. I recommend that you replace the Change Data button with text that instructs the user to change the data in a cell of her choosing. Same thing for the Add Worksheet button: just instruct the user to add a worksheet.